stats/stats/src/charts/lines/new_txns_window.rs

```diff
@@ -0,0 +1,199 @@
+//! New transactions for the last N days (usually 30).
+//!
+//! Allowed to work on a non-indexed networks, as it
+//! recalculates whole N day window/range each time.
+//!
+//! Does not include last day, even as incomplete day.
+
+use crate::{
+    data_source::{
+        kinds::{
+            local_db::{
+                parameters::{
+                    update::batching::{
+                        parameters::{BatchMaxDays, ClearAllAndPassStep},
+                        BatchUpdate,
+                    },
+                    DefaultCreate, DefaultQueryVec,
+                },
+                LocalDbChartSource,
+            },
+            remote_db::{RemoteDatabaseSource, RemoteQueryBehaviour, StatementFromRange},
+        },
+        types::BlockscoutMigrations,
+        UpdateContext,
+    },
+    range::UniversalRange,
+    types::{Timespan, TimespanDuration, TimespanValue},
+    utils::day_start,
+    ChartError, ChartProperties, Named,
+};
+
+use chrono::{DateTime, NaiveDate, Utc};
+use entity::sea_orm_active_enums::ChartType;
+use sea_orm::{FromQueryResult, Statement};
+
+use super::NewTxnsStatement;
+
+pub const WINDOW: u64 = 30;
+
+fn new_txns_window_statement(
+    update_day: NaiveDate,
+    completed_migrations: &BlockscoutMigrations,
+) -> Statement {
+    // `update_day` is not included because the data would
+    // be incomplete.
+    let window =
+        day_start(&update_day.saturating_sub(TimespanDuration::from_timespan_repeats(WINDOW)))
+            ..day_start(&update_day);
+    NewTxnsStatement::get_statement(Some(window), completed_migrations)
+}
+
+pub struct NewTxnsWindowQuery;
+
+impl RemoteQueryBehaviour for NewTxnsWindowQuery {
+    type Output = Vec<TimespanValue<NaiveDate, String>>;
+
+    async fn query_data(
+        cx: &UpdateContext<'_>,
+        _range: UniversalRange<DateTime<Utc>>,
+    ) -> Result<Vec<TimespanValue<NaiveDate, String>>, ChartError> {
+        let update_day = cx.time.date_naive();
+        let query = new_txns_window_statement(update_day, &cx.blockscout_applied_migrations);
+        let mut data = TimespanValue::<NaiveDate, String>::find_by_statement(query)
+            .all(cx.blockscout.connection.as_ref())
+            .await
+            .map_err(ChartError::BlockscoutDB)?;
+        // linear time for sorted sequences
+        data.sort_unstable_by(|a, b| a.timespan.cmp(&b.timespan));
```

Comment:
why sort

Reply:
because I'm not 100% sure that `NewTxnsStatement` will return sorted data